### PR TITLE
JAMES-2309 Long overflow in JMS delays [CONTRIB]

### PR DESCRIPTION
**Pull request by chibenwa**
CI for https://github.com/apache/james-project/pull/125

### CODE REVIEW

**Comment by rouazana** [comment]
```
[4da4077f1fcd73cfeed5c36c06d0f9445731df7c] Failed tests: 
[4da4077f1fcd73cfeed5c36c06d0f9445731df7c]   MailboxManagerStressTest.testStressTest:128 expected:<200> but was:<180>
```
test this please

**Comment by trantienduchn** [comment]
```
[4da4077f1fcd73cfeed5c36c06d0f9445731df7c] java.lang.ArithmeticException: long overflow
[4da4077f1fcd73cfeed5c36c06d0f9445731df7c] 	at org.apache.james.queue.api.DelayedMailQueueContract.enqueueWithVeryLongDelayShouldDelayMail(DelayedMailQueueContract.java:80)
[4da4077f1fcd73cfeed5c36c06d0f9445731df7c] 
[4da4077f1fcd73cfeed5c36c06d0f9445731df7c] 
[4da4077f1fcd73cfeed5c36c06d0f9445731df7c] Results :
[4da4077f1fcd73cfeed5c36c06d0f9445731df7c] 
[4da4077f1fcd73cfeed5c36c06d0f9445731df7c] Tests in error: 
[4da4077f1fcd73cfeed5c36c06d0f9445731df7c]   DelayedMailQueueContract.enqueueWithVeryLongDelayShouldDelayMail:80 » Arithmetic
```

test this please

File: server/queue/queue-api/src/test/java/org/apache/james/queue/api/DelayedMailQueueContract.java [hunk 1]
```diff
@@ -79,7 +79,7 @@ default void enqueueWithReasonablyLongDelayShouldDelayMail(ExecutorService execu
     default void enqueueWithVeryLongDelayShouldDelayMail(ExecutorService executorService) throws Exception {
         getMailQueue().enQueue(defaultMail()
             .build(),
-            100 * 365,
+            Long.MAX_VALUE,
```
Comment (trantienduchn) [inline server/queue/queue-api/src/test/java/org/apache/james/queue/api/DelayedMailQueueContract.java]: Changing delay to `Long.MAX_VALUE` caused `java.lang.ArithmeticException: long overflow` in `MemoryMailQueue::enQueue` then CI build fails. Please take a little fixup from `trantienduchn:JAMES-2309`

Comment (rouazana) [reply server/queue/queue-api/src/test/java/org/apache/james/queue/api/DelayedMailQueueContract.java]: done, thx

File: server/queue/queue-memory/src/main/java/org/apache/james/queue/memory/MemoryMailQueueFactory.java [hunk 2]
```diff
@@ -98,14 +100,26 @@ public String getName() {
...
                 mailItems.put(new MemoryMailQueueItem(cloneMail(mail), this, nextDelivery));
             } catch (MessagingException e) {
                 throw new MailQueueException("Error while copying mail " + mail.getName(), e);
             }
         }
 
+        private ZonedDateTime calculateNextDelivery(long delay, TimeUnit unit) {
+            if (delay > 0) {
+                try {
+                    ZonedDateTime.now().plus(delay, Temporals.chronoUnit(unit));
```
Comment (rouazana) [inline server/queue/queue-memory/src/main/java/org/apache/james/queue/memory/MemoryMailQueueFactory.java]: missing return?

File: server/queue/queue-memory/src/main/java/org/apache/james/queue/memory/MemoryMailQueueFactory.java [hunk 3]
```diff
@@ -98,14 +100,26 @@ public String getName() {
...
+                } catch (ArithmeticException e) {
+                    return Instant.ofEpochMilli(Long.MAX_VALUE).atZone(ZoneId.systemDefault());
```
Comment (rouazana) [inline server/queue/queue-memory/src/main/java/org/apache/james/queue/memory/MemoryMailQueueFactory.java]: I prefer not using system timezones, maybe you can use UTC instead.

File: server/queue/queue-memory/src/main/java/org/apache/james/queue/memory/MemoryMailQueueFactory.java (hunk 2 above)
Comment (trantienduchn) [reply server/queue/queue-memory/src/main/java/org/apache/james/queue/memory/MemoryMailQueueFactory.java]: damn, my bad

File: server/queue/queue-memory/src/main/java/org/apache/james/queue/memory/MemoryMailQueueFactory.java (hunk 3 above)
Comment (trantienduchn) [reply server/queue/queue-memory/src/main/java/org/apache/james/queue/memory/MemoryMailQueueFactory.java]: fixup is available at `trantienduchn/JAMES-2309`

**Comment by chibenwa** [comment]
test this please (no test output)

**Comment by chibenwa** [comment]
Merged

**Pull request closed without merging**